apps/api/lib/api/client/socket.ex

```diff
@@ -8,11 +9,18 @@ defmodule API.Client.Socket do
   ## Authentication
 
   @impl true
-  def connect(%{"token" => token, "id" => external_id}, socket, connect_info) do
-    %{user_agent: user_agent, peer_data: peer_data} = connect_info
+  def connect(%{"token" => token} = attrs, socket, connect_info) do
+    %{user_agent: user_agent, peer_data: %{address: remote_ip}} = connect_info
+
+    # TODO: we want to scope tokens for specific use cases, so token generated in auth flow
```

Comment:
This won't be automatic because we only check expiration when we decode the token, which means that as long as the web socket is alive you can have an indefinite session. What we should do: schedule a websocket termination at the end of token lifetime.